File: Recursion/factorial.cpp
// Factorial of an integer
// Sample Input: 5
// Output: 120
#include<iostream>
using namespace std;
int factorial(int n){
	cout << "I am calculating f(" << n << ")\n";	
	if(n == 0)
		return 1;
	int f =  n * factorial(n - 1);
	cout << "Done! f(" << n << ") = " << f << endl; 
	return f;
}
int main(){
	int n;
	cout << "Give me n : ";
	cin >> n;
	int result = factorial(n);
	cout << result << endl;	
	return 0;
}

File: Recursion/factorial.go
// Factorial of an integer
// Sample Input: 5
// Output: 120

package main

import "fmt"

func Factorial(n int) int {
	if n == 0 {
		return 1;
	}
	return n * Factorial(n - 1)
}

func main() {
	msg := Factorial(5);
	fmt.Println(msg)
}